sources/host-ctr/cmd/host-ctr/pid_namespaces.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/containerd/containerd"
+	"github.com/gofrs/flock"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	// Get file descriptor that refers to a parent namespace, ioctl() request '_IO(NSIO, 0x2)'
+	nsGetParentNs        = 0xb702
+	lockfile      string = "/run/lock/host-ctr-pid-ns.lock"
+	manifestPath  string = "/etc/host-containers/info/pid-namespaces.json"
+)
+
+func getTaskPidNs(task *containerd.Task) (int64, error) {
+	pid := (*task).Pid()
+	pidNsFile, err := os.Open(fmt.Sprintf("/proc/%d/ns/pid", pid))
+	if err != nil {
+		return -1, fmt.Errorf("failed to open %d PID namespace file", pid)
+	}
+	pidNsFd := int(pidNsFile.Fd())
+
+	_, err = unix.IoctlGetInt(pidNsFd, nsGetParentNs)
+	if err == syscall.EPERM {
+		// If we can't access the parent PID namespace, this means the current container task is sharing the root PID
+		// namespace. We don't want to track the root PID namespace in the manifest, so we return -1.
+		return -1, nil
```

Comment:
> the current container task is sharing the root PID namespace. We don't want to track the root PID namespace

I'd like to learn more about this - my intuition is that if the container is using the root PID namespace, that would qualify as a "host" task that we would want to enable third party software to be able to observe.

Reply:
If you include the root PID namespace, then that will include basically all processes on the host and I thought that was one of the requirements; i.e. not to monitor user workloads from orchestrators.

Reply:
Ahhhh, yeah that makes sense 🤔

Would a way to subvert this be to start a host-container that assumes PID 0?